descriptive-link-5effbb.md

```diff
@@ -0,0 +1,237 @@
+
+Note: There may be situations where the purpose of the link is supposed to be unknown or obscured. In those situations, the success criteria will still be met, even if the link's purpose is not described by it.
+
+## Assumptions
+
+_There are currently no assumptions._
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [2.4.4 Link Purpose (In Context)](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html)
+- [G91: Providing link text that describes the purpose of a link](https://www.w3.org/WAI/WCAG21/Techniques/general/G91)
+- [H30: Providing link text that describes the purpose of a link for anchor elements](https://www.w3.org/WAI/WCAG21/Techniques/html/H30)
+- [H24: Providing text alternatives for the area elements of image maps](https://www.w3.org/WAI/WCAG21/Techniques/html/H24)
+- [G53: Identifying the purpose of a link using link text combined with the text of the enclosing sentence](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G53)
+- [ARIA7: Using aria-labelledby for link purpose](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA7)
+- [ARIA8: Using aria-label for link purpose](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA8)
+- [H77: Identifying the purpose of a link using link text combined with its enclosing list item](https://www.w3.org/WAI/WCAG21/Techniques/html/H77)
+- [H78: Identifying the purpose of a link using link text combined with its enclosing paragraph](https://www.w3.org/WAI/WCAG21/Techniques/html/H78)
+- [H79: Identifying the purpose of a link in a data table using the link text combined with its enclosing table cell and associated table header cells](https://www.w3.org/WAI/WCAG21/Techniques/html/H79)
+- [H81: Identifying the purpose of a link in a nested list using link text combined with the parent list item under which the list is nested](https://www.w3.org/WAI/WCAG21/Techniques/html/H81)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+Link text describes the purpose of the link
+
+```html
+<a href="#desc">See the description of this product</a> 
+
+<p id="desc">This product consists of several web pages</p>
+```
+
+#### Passed Example 2
+
+Accessible name describes the purpose of the link
+
+```html
+<a href="#main" aria-label="Go to the main content of the page"></a>
+
+<main>
+ <p id="main">This is the main content</p>
+</main>
+```
+
+#### Passed Example 3
+
+Link text together with its context describe the purpose of the link
+
+```html
+<p>To see the description of this product <a href="#desc">click here</a></p>
+
+<p id="desc">This product consists of several web pages</p>
+```
+
+#### Passed Example 4
+
+Both link text together with its context and accessible name describe the purpose of the link
+
+```html
+<p><a href="#" aria-label="Order item">Place item</a><span aria-hidden="true"> in shopping cart</span></p>
```

Comment:
This would fail "label-in-name" and we should make sure all our examples pass WCAG 2.1 right?